core/src/main/java/org/elasticsearch/index/query/PrefixQueryParser.java

```diff
@@ -74,7 +76,7 @@ public PrefixQueryBuilder fromXContent(QueryParseContext parseContext) throws IO
                     }
                 }
             } else {
-                if (parseContext.parseFieldMatcher().match(currentFieldName, NAME_FIELD)) {
+                if (parseContext.parseFieldMatcher().match(currentFieldName, NAME_FIELD)) {   // this looks weird given that we already parse the _name field above?!
```

Comment:
Re: your comment: left my thoughts on this above, maybe its time to get rid of this version alltogether? Not my call though.